src/config.py

```diff
@@ -55,16 +58,35 @@ def default_config(cls) -> "Config":
             cls()
             .add_receiver(
                 "otlp",
-                {"protocols": {"grpc": {"endpoint": f"0.0.0.0:{Ports.OTLP_GRPC.value}"}}},
+                {"protocols": {"grpc": {"endpoint": f"0.0.0.0:{cls.set_port(PORTS.OTLP_GRPC)}"}}},
                 pipelines=["metrics"],
             )
             .add_exporter(
-                "otlp", {"endpoint": f"otelcol:{Ports.OTLP_GRPC.value}"}, pipelines=["metrics"]
+                "otlp", {"endpoint": f"otelcol:{cls.set_port(PORTS.OTLP_GRPC)}"}, pipelines=["metrics"]
             )
-            .add_extension("health_check", {"endpoint": f"0.0.0.0:{Ports.HEALTH.value}"})
+            .add_extension("health_check", {"endpoint": f"0.0.0.0:{cls.set_port(PORTS.HEALTH)}"})
             .add_telemetry("metrics", "level", "normal")
         )
 
+    @classmethod
+    def active_ports(cls) -> List[int]:
+        """Return the ports that are used in the Collector config."""
+        return list(cls.used_ports)
+
+    @classmethod
+    def clear_ports(cls):
+        """Reset the ports that are used in the Collector config."""
+        cls.used_ports = set()
+
+    @classmethod
+    def set_port(cls, port: int) -> int:
```

Comment:
Traditionally, setters set and getters get.
This reminds a different pattern used sometimes in C, where on success you return the setpoint, and on failure some magic number (e.g. -1).

But I wonder if we need this port at all.
Instead, `active_ports` could perhaps iterate over the current config and extract all indicated ports?